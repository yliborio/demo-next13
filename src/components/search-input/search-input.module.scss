@import "../../styles/variables.scss";

.container {
  display: flex;
  justify-content: center;
  align-items: center;

  .input-text {
    width: 100%;
    background-color: $bg-primary;
    border: none;
    border-radius: 5px;
    padding: 10px 8px;
  }
  > svg {
    transform: translateX(-120%);
  }
}
