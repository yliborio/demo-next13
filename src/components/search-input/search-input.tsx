"use client";
import { useFilterStore } from "../../hooks/useFilter/useFilter";
import { SearchIcon } from "../icons/search-icon";
import styles from "./search-input.module.scss";

export const SearchInput = () => {
  const { setFilter } = useFilterStore();

  return (
    <div className={styles["container"]} data-testid="search-input">
      <input
        placeholder="Search product"
        onChange={(event) => setFilter(event.target.value)}
        className={styles["input-text"]}
      />
      <SearchIcon />
    </div>
  );
};
