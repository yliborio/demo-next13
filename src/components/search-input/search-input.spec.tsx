import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import { SearchInput } from "./search-input";

const mockSetFilter = jest.fn();
jest.mock("../../hooks/useFilter/useFilter", () => {
  return {
    useFilterStore: () => ({
      setFilter: mockSetFilter,
    }),
  };
});

describe("SearchInput", () => {
  it("calls setFilter with the entered search query", () => {
    const { asFragment } = render(<SearchInput />);

    expect(asFragment()).toMatchSnapshot();
    const inputElement = screen.getByPlaceholderText("Search product");

    fireEvent.change(inputElement, { target: { value: "test-query" } });

    expect(mockSetFilter).toHaveBeenCalledWith("test-query");
  });
});
