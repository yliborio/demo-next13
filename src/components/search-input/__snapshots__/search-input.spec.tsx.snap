// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput calls setFilter with the entered search query 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-testid="search-input"
  >
    <input
      class="input-text"
      placeholder="Search product"
    />
    <svg
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15.5 14h-.79l-.28-.27a6.5 6.5 0 1 0-.7.7l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
  </div>
</DocumentFragment>
`;
